r2.8.3

ojw28 opened the pull request.

AquilesCanta committed:
Add license server URL to SchemeData

Allows DrmInitData to carry a license server URL when the media declares one.

Issue:#3393

-------------
Created by MOE: https://github.com/google/moe
MOE_MIGRATED_REVID=199643743

ojw28 committed:
Add isControllerVisible

Issue: #4385

-------------
Created by MOE: https://github.com/google/moe
MOE_MIGRATED_REVID=200986828

andrewlewis committed:
Pass through all ID3 internal data from udta

Also switch from using a CommentFrame to a new InternalFrame type for ID3 data
stored with ID '----', to distinguish internal data from actual ID3 comments.

-------------
Created by MOE: https://github.com/google/moe
MOE_MIGRATED_REVID=201315254

ojw28 committed:
Enable EOS workaround for FireTV Gen 2

-------------
Created by MOE: https://github.com/google/moe
MOE_MIGRATED_REVID=201678261

ojw28 committed:
Add some FireOS workarounds for max input buffer size limitations

-------------
Created by MOE: https://github.com/google/moe
MOE_MIGRATED_REVID=201678686

ojw28 committed:
Extract scheme specific data from PSSH for some FireOS devices

-------------
Created by MOE: https://github.com/google/moe
MOE_MIGRATED_REVID=201679215

AquilesCanta committed:
Parse ms:laurl from ContentProtection in DASH

Issue:#3393

-------------
Created by MOE: https://github.com/google/moe
MOE_MIGRATED_REVID=201694813

ojw28 committed:
Wait for DRM keys before codec configuration on FireOS devices

-------------
Created by MOE: https://github.com/google/moe
MOE_MIGRATED_REVID=201695876

andrewlewis committed:
Make ImaAdsLoader robust to calls after it's released

Issue: #3879

-------------
Created by MOE: https://github.com/google/moe
MOE_MIGRATED_REVID=202100576

AquilesCanta committed:
Rename HlsPlaylistTracker's release to stop

-------------
Created by MOE: https://github.com/google/moe
MOE_MIGRATED_REVID=202103550

tonihei committed:
Add DRM workaround for Asus Zenfone 2.

Issue: #4413

-------------
Created by MOE: https://github.com/google/moe
MOE_MIGRATED_REVID=202277924

murdly committed:
opt out of using DummySurface on specific Fire TV device

keiththompson committed:
Add withSkippedAd method to AdPlaybackState. Currently it is only possible to skip entire ad groups but not individual ads within a given ad group.

tonihei committed:
Add some leeway for finding additional tracks in PsExtractor.

Currently we immediately stop searching after we found one video and one
audio track. This change adds some leeway to detect additional tracks.

Issue:#4406

-------------
Created by MOE: https://github.com/google/moe
MOE_MIGRATED_REVID=202455491

ojw28 committed:
Improve DefaultTrackSelector documentation

It's quite hard to find the defaults currently. Placing them
on each variable makes them easier to find.

-------------
Created by MOE: https://github.com/google/moe
MOE_MIGRATED_REVID=202495929

ojw28 committed:
Simplify DefaultTrackSelector documentation

-------------
Created by MOE: https://github.com/google/moe
MOE_MIGRATED_REVID=202623116

ojw28 committed:
Fix application of styles for CEA-608

Issue: #4321

-------------
Created by MOE: https://github.com/google/moe
MOE_MIGRATED_REVID=202660712

ojw28 committed:
Exclude text streams from duration calculations

Issue: #4029

-------------
Created by MOE: https://github.com/google/moe
MOE_MIGRATED_REVID=202912333

tonihei committed:
Add workaround for unmatched track indices in trex and tkhd.

Both boxes should contain the same list of track indices. However, if only one
track index in each list does not match, we can just assume that these belong
together.

Issue:#4477

-------------
Created by MOE: https://github.com/google/moe
MOE_MIGRATED_REVID=203481258

ojw28 committed:
Avoid providing invalid responses to MediaDrm

MediaDrm.provideXResponse methods only accept the response
corresponding to the most recent MediaDrm.getXRequest call.
Previously, our code allowed the following incorrect call
sequence:

a = getKeyRequest
b = getKeyRequest
provideKeyResponse(responseFor(a));

This would occur in the edge case of a second key request
being triggered whilst the first was still in flight. The
provideKeyResponse call would then fail.

This change fixes the problem by treating responseFor(a)
as stale. Note that a slightly better fix would be to
defer calling getKeyRequest the second time until after
processing the response corresponding to the first one,
however this is significantly harder to implement, and is
probably not worth it for what should be an edge case.

-------------
Created by MOE: https://github.com/google/moe
MOE_MIGRATED_REVID=203481685

tonihei committed:
Add workaround for unmatched track indices in tkhd and tfhd.

If there is only one track, we can assume that both boxes refer to the same track
even if the track indices don't match.

Issue:#4083

-------------
Created by MOE: https://github.com/google/moe
MOE_MIGRATED_REVID=203485872

drhill committed:
flip flag values to their proper names so that trackselector parameters can be useful

ojw28 committed:
Tweak DefaultTrackSelector documentation

Viewport constraints apply to adaptive content even if the
actual playback isn't adaptive (i.e. because the selector
ends up making a fixed track selection).

-------------
Created by MOE: https://github.com/google/moe
MOE_MIGRATED_REVID=203726831

tonihei committed:
Restructure track selection in DashMediaPeriod.

Until now, the streams were released and re-enabled for each type of stream
(primary, event, embedded) in that order. That leads to problems when replacing
streams from one type to another (for example embedded to primary).

This change restructures the track selection to:
1. Release and reset all streams that need to be released or replaced.
 1(a). Including embedded orphan streams.
2. Select new streams.

Issue:#4477

-------------
Created by MOE: https://github.com/google/moe
MOE_MIGRATED_REVID=203751233

tonihei committed:
Fix wrong loop variable in DashMediaPeriod.

-------------
Created by MOE: https://github.com/google/moe
MOE_MIGRATED_REVID=203766579

ojw28 committed:
Move subsampleOffset in Format

It's no longer text specific (it's used for metadata as well, and
in theory could apply to any stream in which samples contain multiple
sub-samples)

-------------
Created by MOE: https://github.com/google/moe
MOE_MIGRATED_REVID=203767825

erdemguven committed:
Fix DownloadService doesn't stop when the app is killed

Also fixed showing "remove notification" when download is completed.

Issue:#4469
Issue:#4488

-------------
Created by MOE: https://github.com/google/moe
MOE_MIGRATED_REVID=203927268

andrewlewis committed:
Ignore all edit lists if one track's edits can't be applied

Issue: #4348

-------------
Created by MOE: https://github.com/google/moe
MOE_MIGRATED_REVID=204261718

botaydotcom committed:
Fix a bug with TTML using font size as % of cellResolution.

After [] we support default font size for TTML, relative to the cellResolution of the document. However, this introduced a bug that makes TTML font-size in such case always follow the cellResolution font size, even when SubtitleView.setApplyEmbeddedStyles(false) and SubtitleView.setApplyEmbeddedFontSizes(false) were used.

This CL updates the fix so that the default font-size using cellResolution works in the same way as other embedded styles, and can be turned off using setters from SubtitleView.

GitHub: #4491

-------------
Created by MOE: https://github.com/google/moe
MOE_MIGRATED_REVID=204467033

ojw28 committed:
Fix bug where sourceId wasn't set for the first chunk

The sample queues haven't been created when the first init call
occurs. In this case we need to set sourceId when we subsequently
create the queues.

-------------
Created by MOE: https://github.com/google/moe
MOE_MIGRATED_REVID=204467538

andrewlewis committed:
Add support for setting companion ad slots

-------------
Created by MOE: https://github.com/google/moe
MOE_MIGRATED_REVID=205373398

tonihei committed:
Fix issue with keeping window sequence number after repeated seeks.

The number is shelved in calls to queue.clear() to keep it for the next
media period. However, the queue may also become empty by repeated calls to
advancePlayingPeriod which may happen when seeking to an unprepared period.

-------------
Created by MOE: https://github.com/google/moe
MOE_MIGRATED_REVID=205376036

tonihei committed:
Update period index in DashMediaPeriod event dispatcher after manifest update.

Issue:#4492

-------------
Created by MOE: https://github.com/google/moe
MOE_MIGRATED_REVID=205636634

ojw28 committed:
Widen setOutputSurface workaround

ojw28 committed:
Fix release branch

ojw28 committed:
Clean up release notes

ojw28 committed:
Update release notes + bump version

-------------
Created by MOE: https://github.com/google/moe
MOE_MIGRATED_REVID=205660355

ojw28 committed:
Add PRO7S to surface switch workaround

Issue: #4468

-------------
Created by MOE: https://github.com/google/moe
MOE_MIGRATED_REVID=205821059

ojw28 committed:
Don't exclude .idea from MOE input

-------------
Created by MOE: https://github.com/google/moe
MOE_MIGRATED_REVID=205988261

erdemguven committed:
Disable automatic imports for inner classes

Inner classes generally has generic names and they make sense when they
are considered with the top level class they appear in.

When this setting is disabled, Android Studio inserts import for the top
level class and append the top level class's name before the reference
to the inner class name.

NO_SQ

-------------
Created by MOE: https://github.com/google/moe
MOE_MIGRATED_REVID=206127763

AquilesCanta committed:
Fix SCTE-35 timestamp adjustment

Issue:#4573

-------------
Created by MOE: https://github.com/google/moe
MOE_MIGRATED_REVID=206737252

erdemguven committed:
Add missing Nullable annotation Player.EventListener.onTimelineChanged

Issue: #4593

-------------
Created by MOE: https://github.com/google/moe
MOE_MIGRATED_REVID=206911927

oleksandr-yefremov commented:
Is there any estimate on when this release might land? We are specifically interested in "Widen setOutputSurface workaround", because we suspect we have negative feedback growing due to it.

ojw28 commented:
We have two more fixes to cherry-pick into the release, both of which are in flight so it should happen soon (i.e. this week).

ojw28 committed:
Apply setOutputSurfaceWorkaround to required FireOS devices

Amazon like to use Device.MODEL, so key on that instead for
these workarounds.

-------------
Created by MOE: https://github.com/google/moe
MOE_MIGRATED_REVID=206917935

ojw28 committed:
Release fixes

ojw28 committed:
Disable tests that fail on some host machines

andrewlewis committed:
Fix DefaultTimeBar accessibility class name

See https://support.google.com/accessibility/android/answer/7661305.

Also fix/suppress nullability warnings.

Issue: #4611

-------------
Created by MOE: https://github.com/google/moe
MOE_MIGRATED_REVID=207245115

ojw28 committed:
Fix freezing playing multi-period DASH with EventStreams

EventSampleStream.readData required eventStreamUpdatable to be
false for END_OF_STREAM to be returned, but it was only being
set to false when the manifest became static. This change also
sets it to false if the period is not the last one in a dynamic
manifest. It also renames it the parameter to appendable, since
that's what it really represents. Events may still be removed
from the start of the EventStream as the period starts moving
out of the live window.

Issue: #4492

-------------
Created by MOE: https://github.com/google/moe
MOE_MIGRATED_REVID=207517265

ojw28 committed:
Minimal fix for non-default cenc name prefixes in DASH manifests

For now, just allow any prefix (and don't check the namespace).
Checking the namespace requires turning on the XmlPullParser
namespace awareness feature, but this can only be done before
parsing, and I think requires us to specify the namespace when
querying other attributes. This would also risk breaking services
who currently don't set namespaces correctly.

It's possible we'll run into similar issues with the few other
tags (e.g. mspr:pro). We should revisit enabling namespace
awareness if/when we see this.

-------------
Created by MOE: https://github.com/google/moe
MOE_MIGRATED_REVID=207526883

ojw28 committed:
Remove Java 8 usage from release

ojw28 committed:
Fix fullscreen icons to be the correct size

ojw28 committed:
Adjust trim silence parameters to make it less aggressive.
